### PR TITLE
Build message inside Postman

### PR DESCRIPTION
**Pull request by thomash-acinq**
The postman now has a higher level API to send messages: the user provides the content and the paths to use and the postman takes care of building the message packet and sending it to the next node.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala [hunk 1]
```diff
@@ -557,30 +556,22 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
...
-              case Postman.SendingStatus(failure: MessageRelay.Failure) => SendOnionMessageResponse(sent = false, Some(failure.toString), None)
-            }
-          case Failure(error) => Future.successful(SendOnionMessageResponse(sent = false, failureMessage = Some(error.toString), response = None))
+    TlvCodecs.tlvStream(MessageOnionCodecs.onionTlvCodec).decode(userCustomContent.bits) match {
+      case Attempt.Successful(DecodeResult(userTlvs, _)) =>
+        val destination = recipient match {
+          case Left(key) => OnionMessages.Recipient(key, None)
+          case Right(route) => OnionMessages.BlindedPath(route)
+        }
+        appKit.postman.ask(ref => Postman.SendMessage(intermediateNodes, destination, replyPath.map(_ :+ appKit.nodeParams.nodeId), userTlvs, ref, appKit.nodeParams.onionMessageConfig.timeout))(timeout, appKit.system.scheduler.toTyped).mapTo[Postman.OnionMessageResponse].map {
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:568]: nit: I would expect the user to include themselves in the `replyPath`: if they do, we will end up duplicating the final `nodeId` by adding it ourselves here. Our current [API documentation](https://acinq.github.io/eclair/#messages) doesn't clarify that, we should update it with the next release. I find it slightly more intuitive to ask the user to include their `nodeId` at the end of the `replyPath` when they provide one, but it's your call!

**Review by t-bast — COMMENTED** [review]
LGTM :+1:

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala:571]: Thanks! But you're now duplicating it in this line, it should be:
```suggestion
        appKit.postman.ask(ref => Postman.SendMessage(intermediateNodes, destination, replyPath, userTlvs, ref, appKit.nodeParams.onionMessageConfig.timeout))(timeout, appKit.system.scheduler.toTyped).mapTo[Postman.OnionMessageResponse].map {
```

Comment (thomash-acinq) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: How did I miss that! It should be fixed now.

**Review by t-bast — APPROVED** [review]

**Pull request merged**